symfony/src/Controller/DeceasedReportsController.php

```diff
@@ -0,0 +1,133 @@
+        $form->handleRequest($request);
+        if ($form->isSubmitted() && $form->isValid()) {
+            $data = $form->getData();
+            try {
+                $fhirData = $deceasedReportsService->buildDeceasedReportReviewFhir($data, $this->getUser());
+                $deceasedReportsService->updateDeceasedReport($participantId, $reportId, $fhirData);
+                $report->setReportStatus('preliminary');
+                $this->addFlash('success', 'Report updated!');
+                return $this->redirectToRoute('deceased_reports_index');
+            } catch (\Exception $e) {
+                error_log($e->getMessage());
+                throw new \Exception('Invalid API response.', $e->getCode());
+            }
+        }
+        return $this->render('deceasedreports/review.html.twig', [
+            'report' => $report,
+            'participant' => $participant,
+            'form' => $form->createView()
+        ]);
+    }
+
+    /**
+     * @Route("/{participantId}/new", name="deceased_report_new", requirements={"participantId"="P\d+"})
+     */
+    public function deceasedReportNew(Request $request, ParticipantSummaryService $participantSummaryService, DeceasedReportsService $deceasedReportsService, $participantId) {
+        $participant = $participantSummaryService->getParticipantById($participantId);
+        if (!$participant) {
+            throw $this->createNotFoundException('Participant not found.');
+        }
+        $reports = $deceasedReportsService->getDeceasedReportsByParticipant($participantId);
+        $report = new DeceasedReport();
+        foreach ($reports as $record) {
+            if ($record->status == 'cancelled') {
+                continue;
+            }
+            $report = (new DeceasedReport())->loadFromFhirObservation($record);
+        }
+        $form = $this->createForm(DeceasedReportType::class, $report, ['disabled' => (bool) $report->getId()]);
+        $form->handleRequest($request);
+        if ($form->isSubmitted() && $form->isValid()) {
+            $report = $form->getData();
+            $form = $this->createForm(DeceasedReportType::class, $report, ['disabled' => true]);
+            try {
+                $fhirData = $deceasedReportsService->buildDeceasedReportFhir($report, $this->getUser());
+                $response = $deceasedReportsService->createDeceasedReport($participantId, $fhirData);
+                $report = $report->loadFromFhirObservation($response);
+                $this->addFlash('success', 'Deceased Report created!');
+            } catch (\Exception $e) {
+                $this->addFlash('error', $e->getMessage());
+            }
+        }
+
+        return $this->render('deceasedreports/new.html.twig', [
+            'participant' => $participant,
+            'report' => $report,
+            'form' => $form->createView()
+        ]);
+    }
+
+    /* Private Methods */
+
+    public function formatReportTableRows($reports = [])
```

Comment:
Why are these not actually private?

Reply:
Should have been. The other method was moved entirely, so it's no longer used. Resolved in c672ddc